### PR TITLE
3 create new note UI

### LINKED ISSUE #3: Create new note UI

### PR DESCRIPTION
**Pull request by abhishekt98**
Fixes #3

### CODE REVIEW

File: app/javascript/src/components/Container.jsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React from "react";
 
-import SideBar from "./sidebar";
+import SideBar from "components/sidebar";
```
Comment (karthiknmenon) [inline app/javascript/src/components/Container.jsx:3]: import SideBar from "components/Sidebar";

File: app/javascript/src/components/Main.jsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import React from "react";
 import { Route, Switch, BrowserRouter as Router } from "react-router-dom";
 import { ToastContainer } from "react-toastify";
 
-import Notes from "./notes";
+import Notes from "components/notes";
```
Comment (karthiknmenon) [inline app/javascript/src/components/Main.jsx:6]: import Notes from "components/Notes";

File: app/javascript/src/components/notes/index.jsx [hunk 3]
```diff
@@ -1,22 +1,23 @@
...
 import Container from "components/Container";
-
-import Deletealert from "./DeleteAlert";
-import Heading from "./Heading";
-import Note from "./note";
+import MenuBar from "components/menubar";
+import DeleteAlert from "components/notes/DeleteAlert";
+import Heading from "components/notes/Heading";
+import Note from "components/notes/note";
+import Addnote from "components/notes/pane/AddNote";
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/index.jsx:8]: Please name files properly.

File: app/javascript/src/components/notes/index.jsx [hunk 4]
```diff
@@ -1,22 +1,23 @@
...
 
 import data from "../../data";
-import MenuBar from "../menubar";
 
 const Index = () => {
   const notes = data.notes;
   const [showDeleteAlert, setShowDeleteAlert] = useState(false);
+  const [showPane, setShowPane] = useState(false);
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/index.jsx:15]: const [isNewNotePaneOpen, setIsNewNotePaneOpen] = useState(false);

File: app/javascript/src/components/notes/pane/AddNote.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+
+import { Pane, Typography } from "neetoui/v2";
+
+import NoteForm from "components/notes/pane/NoteForm";
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/pane/AddNote.jsx:5]: Please refrain from suffixing component or filenames with `Modal, Pane, Form, Page`.

File: app/javascript/src/components/notes/pane/AddNote.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import formInitialValues from "constants/formInitialValues";
+
+const Addnote = ({ showPane, setShowPane }) => {
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/pane/AddNote.jsx:8]: Please name react components properly.

File: app/javascript/src/components/notes/pane/AddNote.jsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  const onClose = () => setShowPane(false);
+
+  return (
+    <Pane isOpen={showPane} onClose={onClose}>
+      <Pane.Header>
+        <Typography style="h2" weight="semibold">
+          Create a New Note
+        </Typography>
+      </Pane.Header>
+      <NoteForm onClose={onClose} note={formInitialValues.notesForm} />
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/pane/AddNote.jsx:18]: No need to suffix `Form` here.

File: app/javascript/src/components/notes/pane/AddNote.jsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+
+import { Pane, Typography } from "neetoui/v2";
+
+import NoteForm from "components/notes/pane/NoteForm";
+import formInitialValues from "constants/formInitialValues";
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/pane/AddNote.jsx:6]: Please move the constant to a constants file in the scope of this directory.
Also, please use a convention as `NOTE_INITIAL_VALUES`.

File: app/javascript/src/components/notes/pane/NoteForm.jsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    Toastr.success("Note has been successfully Added.");
+    onClose();
+  };
+  return (
+    <Formik
+      initialValues={note}
+      onSubmit={handleSubmit}
+      validateOnBlur={submitted}
+      validateOnChange={submitted}
+      validationSchema={formValidationSchemas.notesForm}
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/pane/NoteForm.jsx:22]: Please move this to a constants file in the scope of this directory.

File: app/javascript/src/components/notes/pane/NoteForm.jsx [hunk 10]
```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+
+import { Formik, Form } from "formik";
+import { Button, Pane, Toastr } from "neetoui/v2";
+import { Input, Textarea, Select } from "neetoui/v2/formik";
+
+import formValidationSchemas from "constants/formValidationSchemas";
+
+const NoteForm = ({ onClose, note }) => {
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/pane/NoteForm.jsx:9]: Please remove sufixes.

File: app/javascript/src/components/notes/pane/NoteForm.jsx [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+              />
+
+              <Select
+                name="contact"
+                label="Assigned Contact"
+                placeholder="Select Role"
+                options={[
+                  { label: "option1", value: "value1" },
+                  { label: "option2", value: "value2" },
+                ]}
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/pane/NoteForm.jsx:49]: Please move to a constants file and use proper labels.

File: app/javascript/src/components/notes/pane/NoteForm.jsx [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+                required
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/pane/NoteForm.jsx:50]: Do we need to pass `required` here?

File: app/javascript/src/components/notes/pane/NoteForm.jsx [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+              />
+
+              <Select
+                name="tags"
+                label="Tags"
+                placeholder="Select Tags"
+                options={[
+                  { label: "option1", value: "value1" },
+                  { label: "option2", value: "value2" },
+                ]}
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/pane/NoteForm.jsx:60]: Please move to a constants file and use proper labels.

File: app/javascript/src/components/notes/pane/NoteForm.jsx [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+                required
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/pane/NoteForm.jsx:61]: Do we need to pass required here?

**Review by karthiknmenon — COMMENTED** [review]
@abhishekt98 Please check the suggestions.

File: app/javascript/src/components/notes/pane/NoteForm.jsx (hunk 12 above)
Comment (abhishekt98) [reply app/javascript/src/components/notes/pane/NoteForm.jsx:50]: - In the figma design all the  fields are required field.
Screenshot 2021-12-03 at 4 49 11 PM

**Pull request merged**